Testscripts/Windows/VERIFY-SRIOV-FAILSAFE-XDP.ps1

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Description
+    This script deploys the VM, verifies xdp working with
+    SRIOV-enabled nic and synthetic nic
+#>
+
+param([object] $AllVmData,
+	[object] $CurrentTestData)
+
+$MIN_KERNEL_VERSION = "5.6"
+
+# This function will start ping and xdpdump on client
+function Ping-XDPDump {
+    $VMData = $args[0]
+    $IP = $args[1]
+    $NIC = $args[2]
+    $LogSuffix = $args[3]
+    # Start Ping test
+    $ping_command = "ping -I $NIC -c 30 $IP > ~/pingOut$LogSuffix.txt"
+    Write-LogInfo "Starting command: $ping_command on $($VMData.RoleName)"
+    Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+        -command $ping_command -RunInBackground -runAsSudo
+
+    # Start XDPDump
+    $xdp_command = "cd /root/bpf-samples/xdpdump && timeout 10 ./xdpdump -i $NIC > ~/xdpdumpout$LogSuffix.txt"
+    Write-LogInfo "Starting command: $xdp_command on $($VMData.RoleName)"
+    $testJob = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+        -command $xdp_command -RunInBackground -runAsSudo
+    $timer = 0
+    while ((Get-Job -Id $testJob).State -eq "Running" ) {
+        $currentStatus = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+            -command "tail -2 ~/xdpdumpout$LogSuffix.txt | head -1" -runAsSudo
+        Write-LogInfo "Current Test Status: $currentStatus"
+        Wait-Time -seconds 5
+        $timer += 1
+        if ($timer -gt 15) {
+            Throw "XDPSetup did not stop after 5 mins. Please check logs."
+        }
+    }
+
+    $currentStatus = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password `
+        -command "tail -1 ~/xdpdumpout$LogSuffix.txt | head -1" -runAsSudo
+    if ( ($currentStatus -inotmatch "unloading xdp") ) {
+        Write-LogErr "Test Aborted. Last known status : $currentStatus."
+        Throw "XDP Execution failed"
+    }
+    Write-LogInfo "XDPDump application ran successfully."
+    return "PASS"
+}
+
+function Main {
+    try{
+        $noClient = $true
+		$noServer = $true
+		foreach ($vmData in $allVMData) {
+			if ($vmData.RoleName -imatch "receiver") {
+				$clientVMData = $vmData
+				$noClient = $false
+			}
+			elseif ($vmData.RoleName -imatch "sender") {
+				$noServer = $false
+				$serverVMData = $vmData
+			} else {
+				Write-LogErr "VM role name is not matched with sender or receiver"

```

Comment:
Since you are checking the vm role name here, it will impact to the rest of steps. If you proceed, then what does it happen? $noClient=true, $noServer=true, $clientVMData=null, and $serverVMData=null. It will break and go to catch block. The exception error should be clear. Please review again.

Reply:
Got it!
No need of else block.